napkin_graph.py

```diff
@@ -333,9 +335,13 @@ def create_index_to_json_dict(self) -> 'tuple[dict[int, int], dict[int, int]]':
 		"""
 		id_to_json = {} # Index to JSON Data (node id)
 		json_to_id = {} # JSON Data (node id) to Index
+		counter = 0
 		for i, node in enumerate(self.nodes):
-			id_to_json[i] = node.id
-			json_to_id[node.id] = i
+			id_to_json[counter] = node.id
+			json_to_id[node.id] = counter
```

Comment:
Now that I'm looking at this, maybe a dictionary isn't the right method since we're building 2 dicts to go back and forth. We might want to create some other mechanism (class or data structure) to store this information instead of these 2 dictionaries. Not exactly sure how it needs to be written but I think a good goal to have is eliminating these dictionaries.

Reply:
I mean dictionary does make sense to make the ids to the nodes, I think we need to change the structure of this though